bot/reddit.py

```diff
@@ -13,9 +13,26 @@
     USER_AGENT,
 )
 
-
-_USERNAME_RE = compile(r"/u/[A-Za-z0-9_-]+")
-
+_USERNAME_RE = compile(r"(^|[^\w/])/?u/(?P<name>\w+)")
```

Comment:
Problem with the regex - it's not matching full uris. 

It should match linked user pages, eg `https://reddit.com/u/theelous3` as well as `/u/theelous3`

and optionally also `u/theelous3` but _not_ something like `https://example.com/blahu/theelous3`.

Reply:
new commit catches full usernames